controller-api/src/test/java/com/yahoo/vespa/hosted/controller/api/systemflags/v1/SystemFlagsDataArchiveTest.java

```diff
@@ -260,6 +269,30 @@ public void normalize_json_fail_on_invalid_tenant_id() {
         }
     }
 
+    @Test
+    public void ignores_files_not_related_to_specified_system_definition() {
+        ZoneRegistry registry = createZoneRegistryMock();
+        Path testDirectory = Paths.get("src/test/resources/system-flags-for-multiple-systems/");
+        var archive = SystemFlagsDataArchive.fromDirectoryAndSystem(testDirectory, registry);
+        assertFlagDataHasValue(archive, MY_TEST_FLAG, cdControllerTarget, "default"); // Would be 'cd.controller' if files for CD system were included
+        assertFlagDataHasValue(archive, MY_TEST_FLAG, mainControllerTarget, "default");
+        assertFlagDataHasValue(archive, MY_TEST_FLAG, prodUsWestCfgTarget, "main.prod.us-west-1");
+    }
+
+    @SuppressWarnings("unchecked") // workaround for mocking a method for generic return type
+    private static ZoneRegistry createZoneRegistryMock() {
+        // Cannot use the standard registry mock as it's located in controller-server module
+        ZoneRegistry registryMock = mock(ZoneRegistry.class);
+        when(registryMock.system()).thenReturn(SystemName.main);
+        when(registryMock.getConfigServerVipUri(any())).thenReturn(URI.create("http://localhost:8080/"));
+        when(registryMock.getConfigServerHttpsIdentity(any())).thenReturn(new AthenzService("domain", "servicename"));
+        ZoneList zoneListMock = mock(ZoneList.class);
+        when(zoneListMock.reachable()).thenReturn(zoneListMock);
+        when(zoneListMock.zones()).thenReturn((List)List.of(new SimpleZone("prod.us-west-1"), new SimpleZone("prod.us-east-3")));
```

Comment:
Consider replacing SimpleZone with (a method that defines) a mock(ZoneApi.class) that has `when()` defined on getSystemName() and getId()?

Reply:
I don't have a strong opinion about this, using Mockito for all mocks looks more consistent. As Mockito is somewhat controversial among Vespa devs, I try to not use it unless it's the superior choice.